litex/tools/litex_sim.py

```diff
@@ -62,6 +62,72 @@ class Platform(SimPlatform):
     def __init__(self):
         SimPlatform.__init__(self, "SIM", _io)
 
+# DFI PHY model settings ---------------------------------------------------------------------------
+
+sdr_phy_settings = {
+    "nphases": 1,
+    "rdphase": 0,
+    "wrphase": 0,
+    "rdcmdphase": 0,
+    "wrcmdphase": 0,
+    "cl": 2,
+    "read_latency": 4,
+    "write_latency": 0
+}
+
+sdram_module_rates = {
+    "SDR": "1:1",
+    "DDR": "1:2",
+    "LPDDR": "1:2",
+    "DDR2": "1:2",
+    "DDR3": "1:4",
+}
+
+def get_sdram_phy_settings(memtype, data_width, clk_freq):
+    nphases = 4 if memtype == "DDR3" else 2
```

Comment:
`rates`/`nphases` values always match so you can probably also reuse `sdram_module_rates` dict.

Reply:
Good idea, I'm now using single dict for `rates/nphases`